### PR TITLE
Split setting page in 3 page, moving existing feild to new tab

### PR DESCRIPTION
**Pull request by lalit8347**
(amplication-client)  App setting is divided in to 3 tabs, Db setting , App Setting and Auth Setting.

### CODE REVIEW

File: packages/amplication-client/src/Application/ApplicationSettingsForm.tsx [hunk 1]
```diff
@@ -47,7 +47,9 @@ const FORM_SCHEMA = {
 
 const CLASS_NAME = "application-settings-form";
 
-function ApplicationSettingsForm({ applicationId }: Props) {
+function ApplicationSettingsForm({ match }: Props) {
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Application/ApplicationSettingsForm.tsx:50]: rename to ApplicationDatabaseSettingsForms

File: packages/amplication-client/src/Application/ApplicationForm.tsx [hunk 2]
```diff
@@ -15,7 +15,7 @@ import { COLORS } from "./constants";
 import { ColorSelectButton } from "../Components/ColorSelectButton";
 import { useTracking } from "../util/analytics";
 import { GET_APPLICATION } from "./ApplicationHome";
-import ApplicationSettingsForm from "./ApplicationSettingsForm";
+// import ApplicationSettingsForm from "./ApplicationSettingsForm";
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Application/ApplicationForm.tsx:18]: remove comment

File: packages/amplication-client/src/Application/ApplicationForm.tsx [hunk 3]
```diff
@@ -121,8 +121,8 @@ function ApplicationForm({ match }: Props) {
               );
             }}
           </Formik>
-          <hr />
-          <ApplicationSettingsForm applicationId={applicationId} />
+         
+          {/* <ApplicationSettingsForm applicationId={applicationId} /> */}
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Application/ApplicationForm.tsx:125]: remove comment

File: packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      
+              onSubmit={() => { }}
+        >
+          {(formik) => {
+            return (
+              <Form>
+                <h3>Authentication Providers</h3>
+               
+                
+                    <TextField name="dbHost" autoComplete="off" label="Default Provider" value={ "Basic HTTP"}/>
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx:32]: This should be a dropdown with the available values

File: packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                  <ToggleField
+                    name="http"
+                    label="Basic HTTP"
+                    />
+                    <br />
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx:37]: remove <br/>

File: packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+                    label="JWT"
+              />
+                
+                    <hr />
+
+                    <h3>Default Credentials</h3>
+              
+            
+                <TextField
+                  name="dbName"
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx:49]: when connecting to the server, use the correct field

File: packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+                  autoComplete="off"
+                  label="App Default User Name"
+                />
+                <TextField
+                  name="dbPort"
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx:54]: when connecting to the server, use the correct field

File: packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+import { Form, Formik } from "formik";
+import React from "react";
+
+import { match } from "react-router-dom";
+import "./ApplicationForm.scss";
+type Props = {
+  match: match<{ application: string }>;
+};
+
+const CLASS_NAME = "application-form";
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx:13]: rename to application-database-settings-form

File: packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx [hunk 9]
```diff
@@ -0,0 +1,69 @@
+import { TextField, ToggleField } from "@amplication/design-system";
+
+import "@rmwc/snackbar/styles";
+import { Form, Formik } from "formik";
+import React from "react";
+
+import { match } from "react-router-dom";
+import "./ApplicationForm.scss";
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx:8]: replace to ApplicationDatabaseSettingsForm.scss

File: packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+function ApplicationAuthSettingForm({ match }: Props) {
+
+  return (
+    <div className={CLASS_NAME}>
+      
+        <Formik
+      
+              initialValues={{}}
+      
+              onSubmit={() => { }}
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx:24]: after connecting to the server add the missing initialValues and onSubmit

File: packages/amplication-client/src/models.ts [hunk 11]
```diff
@@ -2,6 +2,10 @@ export type Maybe<T> = T | null;
 export type Exact<T extends { [key: string]: unknown }> = {
   [K in keyof T]: T[K];
 };
+export type MakeOptional<T, K extends keyof T> = Omit<T, K> &
+  { [SubKey in K]?: Maybe<T[SubKey]> };
+export type MakeMaybe<T, K extends keyof T> = Omit<T, K> &
+  { [SubKey in K]: Maybe<T[SubKey]> };
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/models.ts:8]: I am not sure where this changes originated from and why it was required to generate the files at all

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]
Well done!
Please see my comments. 
If you want we can schedule a time to discuss

**Comment by lalit8347** [comment]
require changes added.

File: packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx [hunk 12]
```diff
@@ -5,40 +5,65 @@ import { Form, Formik } from "formik";
...
+                    formik.setFieldValue("http", true);
+                    formik.setFieldValue("jwt", false);
+                  }
+                  else {
+                    formik.setFieldValue("jwt", true);
+                    formik.setFieldValue("http", false);
+                  }
+                  formik.handleChange(e)
+                }
+                }>
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx:47]: use useCallback

File: packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx [hunk 13]
```diff
@@ -5,40 +5,65 @@ import { Form, Formik } from "formik";
...
+                  if (e.target.value === "http") {
+                    formik.setFieldValue("http", true);
+                    formik.setFieldValue("jwt", false);
+                  }
+                  else {
+                    formik.setFieldValue("jwt", true);
+                    formik.setFieldValue("http", false);
+                  }
+                  formik.handleChange(e)
+                }
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx:46]: The logic should be a little different. 
Let's approve this PR and talk about the logic in the next PR when we connect the server

File: packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx [hunk 14]
```diff
@@ -5,40 +5,65 @@ import { Form, Formik } from "formik";
...
+                <ToggleField
                     name="http"
                     label="Basic HTTP"
-                    />
-                    <br />
+                    disabled={!false}
+                    
+                  />
+                  </div>
+                  <div style={{marginTop :"10px"}}>
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx:59]: don't use inline style

File: packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx [hunk 15]
```diff
@@ -5,40 +5,65 @@ import { Form, Formik } from "formik";
...
+                    formik.setFieldValue("jwt", true);
+                    formik.setFieldValue("http", false);
+                  }
+                  formik.handleChange(e)
+                }
+                }>
+                  <option selected value="http">Basic HTTP</option>
+                  <option value="jwt">JWT</option>
+                </select>
+                <div style={{marginTop :"10px"}}>
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Application/ApplicationAuthSettingForm.tsx:51]: There is no need for the wrapper div.
the ToggleField should handle all the styling

File: packages/amplication-client/src/Application/ApplicationDatabaseSettingsForms.tsx [hunk 16]
```diff
@@ -45,9 +45,9 @@ const FORM_SCHEMA = {
   },
 };
 
-const CLASS_NAME = "application-settings-form";
+const CLASS_NAME = "ApplicationDatabaseSettingsForm.scss";
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Application/ApplicationDatabaseSettingsForms.tsx:48]: this should be kebab-case without the .scss

**Review by yuval-hazaz — APPROVED** [review]
I approve this PR since all the required changes are part of the new form which anyway is not connected to the server yet.
In the next PR, when we add the connection to the server, please handle the new comments

**Pull request merged**